build-system/tasks/build.js

```diff
@@ -90,6 +91,7 @@ async function performBuild(watch) {
   await performPrerequisiteSteps(watch);
   await compileAllUnminifiedJs(watch);
   await buildExtensions({watch});
+  await formatExtractedMessages();
```

Comment:
Assuming log message extraction doesn't apply to extensions, moving `formatExtractedMessages()` to the line after `compileAllUnminifiedJs()` will greatly reduce wait time if you're testing this stuff during a full build.

Reply:
Also, does this have to be re-done during `watch` if the messages change? (Can do this in in a separate PR.)

Reply:
It does apply to extensions and yea, they have to be regenerated on `watch`.

This would be a NO-OP if the message table doesn't change, but if it does then at least one message ID reference would be orphaned and therefore break logging if triggered.

Reply:
(I thought the steps I took here applied to `watch`. LMK where I should put this if not.)

Reply:
> I thought the steps I took here applied to `watch`

This PR as currently written doesn't work with extensions or with `watch`. I think it makes sense for this to be agnostic of lazy building vs. building everything, and minified vs. unminified. Will follow up with a set of suggestions for how to make it so.

Reply:
Much appreciated.

Reply:
Chatted offline with @alanorozco to get a better understanding of how log extraction works.

- For `gulp dist` (minified), there's already a call to `formatExtractedMessages()` right at the end of `dist()` in `build-system/tasks/dist.js` (This can be left as is, since we don't support lazy building with `gulp dist`.)
- For`gulp build`, `gulp watch`, and `gulp` (all unminified), a new `format*` function that acts only on the file being built can be called at the end of `compileMinifiedJs` in `build-system/tasks/helpers.js` (via an additonal `.then()` step). This will work on extensions and non-extension JS files alike, and will automatically be picked up during `watch` and during lazy building.